modules/controller/index.js

```diff
@@ -1,16 +1,138 @@
 const updateResponse = require('./lib/responders/update');
 const destroyResponse = require('./lib/responders/destroy');
-const lookupFailed = require('./lib/responders/lookup_failed');
+const lookupFailedResponse = require('./lib/responders/lookup_failed');
+
+const filterInitValidator = require('./lib/validators/init/filters');
+const relationsInitValidator = require('./lib/validators/init/relations');
+const modelMethodInitValidator = require('./lib/validators/init/model_method');
+const relationExistsReqValidator = require('./lib/validators/request/relation_exists');
+const contentTypeReqValidator = require('./lib/validators/request/content_type');
 
 function Controller(opts) {
   _.extend(this, parseOptions(opts));
 }
 
+Controller.prototype._configureMethod = function (method, opts) {
+  var requestHandler = this._buildRequestHandler.bind(this);
+  var validateInit = this._validateInit.bind(this);
+  var optsOrDefault = this._optsOrDefault.bind(this);
+
+  opts = optsOrDefault(opts, method);
+
+  validateInit(opts);
+
+  return requestHandler(opts);
+};
+
+Controller.prototype._optsOrDefault = function(opts, method) {
+  var source = this.source;
+  return _.defaults({}, opts, {
+    include: [],
+    filters: {},
+    type: source.typeName(),
+    modelMethod: opts && opts.method ? opts.method : method,
+    controllerMethod: method
+  });
+};
+
+Controller.prototype._validateInit = function(opts) {
+  var source = this.source;
+
+  filterInitValidator(opts, source);
+  relationsInitValidator(opts, source);
+  modelMethodInitValidator(opts, source);
+};
+
+Controller.prototype._buildRequestHandler = function(opts) {
+  var source = this.source;
+  var responder = opts.responder || this.responder;
+  var method = opts.controllerMethod;
+
+  var validateRequest = this._lookupValidateRequest(method);
+  var polyResponse = this._lookupPolyResponse(method);
+  var handle = this._lookupHandler(method).bind(this);
+
+  return function(request, response, next) {
+    var err = validateRequest(request, response, source);
+
+    if (err) {
+      return responder(polyResponse(err), request, response);
+    }
+
+    handle(opts, request, response, next);
+  };
+};
+
+Controller.prototype._lookupValidateRequest = function(method) {
+  switch (method) {
+    case 'readRelation':
+      return relationExistsReqValidator;
+    case 'create':
+    case 'update':
+      return contentTypeReqValidator;
+    default:
+      return function() {};
+  }
+};
+
+Controller.prototype._lookupPolyResponse = function(method) {
+  switch (method) {
+    case 'read':
+    case 'readRelation':
+      return readResponse;
+    case 'create':
+      return createResponse;
+    case 'update':
+      return updateResponse;
+    case 'destroy':
+      return destroyResponse;
+    case 'lookupFailed':
+      return lookupFailedResponse;
+    default:
+      return readResponse;
+  }
+};
+
+Controller.prototype._lookupHandler = function(method) {
```

Comment:
Can we replace this switch statement with something along the lines of `this['_'+method+'Handler']`?